File: backend/shared/audit.js
function buildAuditRecord({
  sessionId,
  userRole,
  queryText,
  route,
}) {
  return {
    session_id: sessionId,
    user_role: userRole,
    query_text: queryText,
    query_mode: route.mode,
    intent: route.intent,
    generated_zcql: route.zcql,
    template_id: route.templateId,
    source_reference: route.sourceReference,
    explainability_version: "v1-node",
    created_at: new Date().toISOString(),
  };
}

module.exports = {
  buildAuditRecord,
};

File: backend/shared/audit.py
from datetime import datetime, timezone


def build_audit_record(
    session_id,
    user_role,
    query_text,
    query_mode,
    generated_sql=None,
    source_reference=None,
):
    return {
        "session_id": session_id,
        "user_role": user_role,
        "query_text": query_text,
        "query_mode": query_mode,
        "generated_sql": generated_sql,
        "source_reference": source_reference,
        "explainability_version": "v1",
        "created_at": datetime.now(timezone.utc).isoformat(),
    }
